File: 13-roman-to-integer/roman-to-integer.c
int romanToInt(char* s) {
    int sum = 0; // Initialize sum to 0
    int i = 0;
    
    while (s[i] != '\0') {
        if (s[i] == 'M') {
            sum += 1000;
            i++;
        }
        else if (s[i] == 'D') {
            sum += 500;
            i++;
        }
        else if (s[i] == 'C' && s[i + 1] == 'M') {
            sum += 900;
            i += 2;
        }
        else if (s[i] == 'C' && s[i + 1] == 'D') {
            sum += 400;
            i += 2;
        }
        else if (s[i] == 'C') {
            sum += 100;
            i++;
        }
        else if (s[i] == 'L') {
            sum += 50;
            i++;
        }
        else if (s[i] == 'X' && s[i + 1] == 'C') {
            sum += 90;
            i += 2;
        }
        else if (s[i] == 'X' && s[i + 1] == 'L') {
            sum += 40;
            i += 2;
        }
        else if (s[i] == 'X') {
            sum += 10;
            i++;
        }
        else if (s[i] == 'V') {
            sum += 5;
            i++;
        }
        else if (s[i] == 'I' && s[i + 1] == 'X') {
            sum += 9;
            i += 2;
        }
        else if (s[i] == 'I' && s[i + 1] == 'V') {
            sum += 4;
            i += 2;
        }
        else if (s[i] == 'I') {
            sum += 1;
            i++;
        }
    }
    return sum;
}

File: 13-roman-to-integer/roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        count = 0
        i = 0
        while i < len(s):
            if i + 1 < len(s) and s[i:i+2] in ["IV", "IX", "XL", "XC", "CD", "CM"]:
                if s[i:i+2] == "IV":
                    count += 4
                elif s[i:i+2] == "IX":
                    count += 9
                elif s[i:i+2] == "XL":
                    count += 40
                elif s[i:i+2] == "XC":
                    count += 90
                elif s[i:i+2] == "CD":
                    count += 400
                elif s[i:i+2] == "CM":
                    count += 900
                i += 2   # skip next char since it’s already counted
            else:
                if s[i] == 'I':
                    count += 1
                elif s[i] == 'V':
                    count += 5
                elif s[i] == 'X':
                    count += 10
                elif s[i] == 'L':
                    count += 50
                elif s[i] == 'C':
                    count += 100
                elif s[i] == 'D':
                    count += 500
                elif s[i] == 'M':
                    count += 1000
                i += 1
        return count

            
            

                

        